src/Command/WorkerCommand.php

```diff
@@ -96,45 +99,46 @@ protected function getQueueExtension(LoggerInterface $logger): QueueExtension
     /**
      * Creates and returns a LoggerInterface object
      *
+     * @param Arguments $args
      * @return \Psr\Log\LoggerInterface
      */
-    protected function getLogger(): LoggerInterface
+    protected function getLogger(Arguments $args): LoggerInterface
     {
         $logger = new NullLogger();
-        if (!empty($this->params['verbose'])) {
-            $logger = Log::engine($this->params['logger']);
+        if (!empty($args->getOption('verbose'))) {
+            $logger = Log::engine($args->getOption('logger'));
         }
 
         return $logger;
     }
 
     /**
-     * main() method.
-     *
-     * @return null
+     * @param  Arguments  $args
+     * @param  ConsoleIo  $io
+     * @return int|void|null
      */
-    public function main()
+    public function execute(Arguments $args, ConsoleIo $io)
     {
-        $logger = $this->getLogger();
+        $logger = $this->getLogger($args);
         $processor = new Processor($logger);
-        $extension = $this->getQueueExtension($logger);
+        $extension = $this->getQueueExtension($args, $logger);
+
+        $config = $args->getOption('config');
+        $listenerClassName = Configure::read(sprintf('Queue.%s.listener', $config));
```

Comment:
I don't think we should do this, or fail the execution unless the `config` option was used. Listeners are not required to process jobs.

Reply:
I tried testing for a scenario that would cause this to die, I couldn't get one other than if the config key including default was entirely gone. So I put in some more checks including one that checks for the existence of the key in the config before it is used to get the url or the listener class. What do you think of the messaging in the errors?

Reply:
Looks good to me.